lib/companies_house/request.rb

```diff
@@ -0,0 +1,94 @@
+# frozen_string_literal: true
+require 'companies_house/api_error'
+require 'companies_house/not_found_error'
+require 'companies_house/authentication_error'
+require 'companies_house/rate_limit_error'
+
+require 'virtus'
+require 'uri'
+require 'json'
+
+require 'active_support/notifications'
+
+module CompaniesHouse
+  # This class manages individual requests.
+  # Users of the CompaniesHouse gem should not instantiate this class
+  # and should instead use CompaniesHouse::Client.
+  class Request
```

Comment:
The `.request` method was exceeding the ABC limit in Rubocop. This new class should help. It's extracted all the requesty logic, leaving the Client managing HTTP connections and doing "loop over a few requests" operations. (For the moment parsing is a feature of the request as well, since it's just `JSON[]`).